subsys/bluetooth/audio/vcs_client.c

```diff
@@ -75,6 +75,21 @@ static struct vcs_instance *lookup_vcs_by_vocs(const struct bt_vocs *vocs)
 	return NULL;
 }
 
+static struct vcs_instance *lookup_vcs_by_aics(const struct bt_aics *aics)
```

Comment:
Seeing this in many of the Audio services, would it make sense to cache the pointer instead?
You might already have considered that, so I'd be interested to know the reason for not doing it.

Reply:
By caching, you mean storing a reference to e.g. the `struct bt_vcs *vcs` in the `aics` instance?

Since we don't know which primary service includes AICS, we would have to store a pointer to the "parent" using a void pointer or something. It's worth considering to cache the parent pointer. 

Do you mind if I take this into consideration and possibly add later after the API changes has been merged?

Reply:
I mean storing a reference, yes. It's something that l2cap and users of l2cap does for example.

Just seemed like something that might be inefficient, but I'm not sure how often these would be called and if the trade-off is worth it or not.

Yeah, I don't mind doing this later, or not at all if you think that is best.

Reply:
lookup_vcs_by_aics is actually only used once (not sure about the others) in the discovery callback (which will rarely be called) , so it will be a simple tradeoff between code size and RAM. Any performance difference will be negligible. 

Of course if we include AICS many places (each needing such a function) a simple reference will be much better. 

I'll note it down :)